balancer-js/src/modules/sor/pool-data/onChainData3.ts

```diff
@@ -0,0 +1,230 @@
+  'function getScalingFactors() view returns (uint256[] memory)', // do we need this here?
+  'function getPoolTokens(bytes32) view returns (address[], uint256[])',
+];
+
+const getTotalSupplyFn = (poolType: string) => {
+  if (poolType.includes('Linear') || ['StablePhantom'].includes(poolType)) {
+    return 'getVirtualSupply';
+  } else if (poolType === 'ComposableStable') {
+    return 'getActualSupply';
+  } else {
+    return 'totalSupply';
+  }
+};
+
+const getSwapFeeFn = (poolType: string) => {
+  if (poolType === 'Element') {
+    return 'percentFee';
+  } else if (poolType === 'FX') {
+    return 'protocolPercentFee';
+  } else {
+    return 'getSwapFeePercentage';
+  }
+};
+
+interface OnchainData {
+  poolTokens: [string[], string[]];
+  totalShares: string;
+  swapFee: string;
+  isPaused?: boolean;
+  inRecoveryMode?: boolean;
+  rate?: string;
+  scalingFactors?: string[];
+  weights?: string[];
+  targets?: [string, string];
+  wrappedTokenRate?: string;
+  amp?: [string, boolean, string];
+  tokenRates?: [string, string];
+}
+
+const defaultCalls = (
+  id: string,
+  address: string,
+  vaultAddress: string,
+  poolType: string,
+  multicaller: Multicaller3
+) => {
+  multicaller.call(`${id}.poolTokens`, vaultAddress, 'getPoolTokens', [id]);
+  multicaller.call(`${id}.totalShares`, address, getTotalSupplyFn(poolType));
+  multicaller.call(`${id}.swapFee`, address, getSwapFeeFn(poolType));
+  // Following where added to the pools query contract, however legacy onchain data didn't have them.
+  // multicaller.call(`${id}.isPaused`, address, 'getPausedState');
+  // multicaller.call(`${id}.inRecoveryMode`, address, 'inRecoveryMode');
+  // multicaller.call(`${id}.rate`, address, 'getRate');
+  // multicaller.call(`${id}.scalingFactors`, address, 'getScalingFactors');
+};
+
+const weightedCalls = (
+  id: string,
+  address: string,
+  multicaller: Multicaller3
+) => {
+  multicaller.call(`${id}.weights`, address, 'getNormalizedWeights');
+};
+
+const linearCalls = (
+  id: string,
+  address: string,
+  multicaller: Multicaller3
+) => {
+  multicaller.call(`${id}.targets`, address, 'getTargets');
+  multicaller.call(`${id}.wrappedTokenRate`, address, 'getWrappedTokenRate');
+};
+
+const stableCalls = (
+  id: string,
+  address: string,
+  multicaller: Multicaller3
+) => {
+  multicaller.call(`${id}.amp`, address, 'getAmplificationParameter');
+};
+
+const gyroECalls = (id: string, address: string, multicaller: Multicaller3) => {
+  multicaller.call(`${id}.tokenRates`, address, 'getTokenRates');
+};
+
+const poolTypeCalls = (poolType: string, poolTypeVersion = 1) => {
+  const do_nothing = () => ({});
+  switch (poolType) {
+    case 'Weighted':
+    case 'LiquidityBootstrapping':
+    case 'Investment':
+      return weightedCalls;
+    case 'Stable':
+    case 'StablePhantom':
+    case 'MetaStable':
+    case 'ComposableStable':
+      return stableCalls;
+    case 'GyroE':
+      if (poolTypeVersion > 1) {
```

Comment:
Do you think its better to use `poolTypeVersion === 2` as thats the case we know for certain or use `poolTypeVersion > 1` which maybe break future versions (there are no versions >2 so far)?